core/lib/instrument_api.h

```diff
@@ -1806,6 +1806,32 @@ DR_API
 const char *
 dr_get_application_name(void);
 
+DR_API
+/**
+ * Provides information about the app's arguments by setting \p args_buf up to
+ * the size \p buf_size. Returns the number of args set or -1 on error.
```

Comment:
We should provide a way for the user to know how many args there are so it can resize and try again.  Should it return the total # of args even if that's more than the passed-in?  Or at least provide a specific error reason so the caller knows whether to retry w/ a bigger buffer or give up.

Reply:
I think this should be a separate API function: dr_num_app_args()

Reply:
Done.